react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+
+function Circles(props) {
+  const circles = props.pictures.map(picture => {
+    const id = picture.id;
+    return (
+      <div
+        key={picture.id}
+        className={props.pictureId === id ? 'circle-outline filled' : 'circle-outline'}
+        onClick={props.pictureId < id ? props.handleNextIconClick : props.handleBackIconClick}
+      ></div>
+    );
+  });
+  return circles;
+}
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      pictureId: 1,
+      pictures: []
```

Comment:
In this situation, i recommend basing your state on an index instead of an id. if picture #1 gets deleted for some reason, it'll break your component, which should be generic enough to not care.